### PR TITLE
Admin middleware

### PR DESCRIPTION
**Pull request by Bui-Cong-Tri**
## Related Tickets:
- [#Ticket ID](https://edu-redmine.sun-asterisk.vn/issues/66505)

## WHAT this PR does?
Thêm admin middleware và enum cho role

## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

**Comment by Bui-Cong-Tri** [comment]
Ready

**Comment by manhtd-1083** [comment]
conflict.

**Comment by ghost** [comment]
conflicted

**Comment by Bui-Cong-Tri** [comment]
Ready

File: app/Http/Middleware/CheckAdmin.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+     * @param  \Closure(\Illuminate\Http\Request): (\Illuminate\Http\Response|\Illuminate\Http\RedirectResponse)  $next
+     * @return \Illuminate\Http\Response|\Illuminate\Http\RedirectResponse
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        if (Auth::user()->role === UserRole::ROLE_ADMIN) {
+            return $next($request);
+        }
+
+        return redirect('/dashboard')->with('fail', 'You don\'t have permission to access this resources!');
```
Comment (ghost) [inline app/Http/Middleware/CheckAdmin.php]: localization

**Comment by Bui-Cong-Tri** [comment]
Ready

**Review by ghost — APPROVED** [review]

**Pull request merged**